### ISSUE TITLE
New typescript-eslint rule `no-useless-default-assignment` false positive on Svelte files

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#1447]
Hi,
typescript-eslint added a new rule in v8.50, also enabled in the strict-type-checked config. However, this rule has false positives for svelte bindable props - see e. g. scout-handbook/handbook-web-admin#1624.

What should we do? Do we need to override this rule here?

**Comment by KieranP** [comment]
**Edit:** Sorry, I see you referred to $bindable() props, and you are correct, it is warning and there is no good fix for that. My comment below is irrelevant, please ignore.


I thought the same too, but it does seem to still work. Just tried this in the svelte playground:

```svelte
<script>
  import Child from './Component.svelte'
</script>

<Child />
```

```svelte
<script>
  const { heading = undefined } = $props()
  $inspect(heading)
</script>

{#if heading}
  <h1>{heading}</h1>
{:else}
  <h1>No Heading</h1>
{/if}
```

If I remove the `= undefined` part of props, it all still functions as expected:

```svelte
<Child /> <- works. shows "No Heading"
<Child heading={undefined} /> <- works, shows "No Heading"
<Child heading="Hello" /> <- works, shows "Hello"
```

Evidently, $props returns an undefined by default when the prop isn't passed into the component, so assigning `heading` when `heading` is not in $props  is already the same as `heading = undefined`.

So it appears that the new TypeScript rule is correct, in the above case anyway. There is still the discussion though of whether `= undefined` provides clearer intent for that variable and whether it should stay.

**Comment by notnotjake** [comment]
Also running into this and would like to work on a fix. I think that a post process hook could check if the line is a $bindable prop assignment and could drop the report.

**Comment by qRoC** [comment]
Example:

```typescript
  let { ref = $bindable(null) }: { ref: HTMLElement | null; } = $props();
```

eslint autofix to:

```typescript
  let { ref }: { ref: HTMLElement | null; } = $props();
```
